src/WidgetBase.ts

```diff
@@ -156,6 +124,20 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 		this.own(this._nodeHandler);
 		this._boundRenderFunc = this.render.bind(this);
 		this._boundInvalidate = this.invalidate.bind(this);
+		this.own(this.on('element-created', ({ key, element }: any) => {
```

Comment:
you can use a map here, which returns a single composite handle:

```
this.on({
  'element-create'({ key, element }: any) { /* ... */ },
  'eleement-updated'({ key, element: any) { /* ... */ },
  ...
});
```

Also it seems like the `any` could/should be more specific?

Reply:
Nice I'll give that a try, also add the proper typings over `any`.

Reply:
Updated